scripts/views/projectView.js

```diff
@@ -1,15 +1,45 @@
-var handleMainNav = function() {
-  $('.main-nav').on('click', '.nav-item', function(e) {
-    $('.tab-content').hide();
-    $(`#${$(this).data('content')}`).fadeIn(500);
-  });
-  $('.main-nav .nav-item:first').click();
-};
-
-var renderProjects = function() {
-  Project.allProjects.forEach(function(project) {
-    $('#projects').append(project.toHtml());
-  });
-};
-handleMainNav();
-renderProjects();
+(function(module) {
+
+  var projectView = {};
+
+  projectView.handleMainNav = function() {
+    $('.main-nav').on('click', '.nav-item', function(e) {
+      $('.tab-content').hide();
+      $('select').val('');
+      $('article').fadeIn(500);
+      $(`#${$(this).data('content')}`).fadeIn(500);
+    });
+    $('.main-nav .nav-item:first').click();
+  };
+
+  projectView.handleDifficultyFilter = function() {
+    $('#difficulty-filter').on('change', function() {
+      if ($(this).val()) {
+        $('article').hide();
+        $('article[data-difficulty="' + $(this).val() + '"]').fadeIn(500);
+      }
+      else {
+        $('article').fadeIn(500);
+      }
+    })
+  }
+
+  projectView.renderProjects = function() {
+    var projectsByDiff = Project.byDifficulty();
+    projectsByDiff.forEach(function(p) {
+      if ($('#difficulty-filter option:contains("' + p.difficulty + '")').length === 0) {
+        $('#difficulty-filter').append(p.toHtml());
+        $('#projects').append(p.toHtml());
```

Comment:
Nice work rendering your projects to the page. Is there a reason that you're messing with #difficulty-filter here? If you take out all of that logic, your app functions essentially the same way. As your difficulty filter options are pre-populated, all you have to do is handle when one is selected (on projectView.handleDifficultyFilter).

```
  projectView.renderProjects = function() {
    var projectsByDiff = Project.byDifficulty();
    projectsByDiff.forEach(function(p) {
        $('#projects').append(p.toHtml());
    });
  };
```